[Windows] Add new setting "Use 10 bit for SDR" in Display Settings

thexai opened the pull request:
## Description
- Add new setting "Use 10 bit for SDR" in Display Settings.
- Removes advanced setting `try10bitoutput` as the new setting already provides this functionality (and more).

## Motivation and context
This functionality has been around for a long time but it was only accessible through advanced settings so it is surely little used. Also now in the master branch and even v19.2 is also used by default when the system supports HDR pass-through (since all systems that support HDR must also support 10-bit).

However that leaves out users who do not have an HDR display but have modern systems that are perfectly capable of 10-bit.

Now this setting simply makes this possibility more visible and provides more fine control:

**[Auto detect] (default)**: It does not changes the current behavior, 10-bit swapchain is used (also for SDR content) if an HDR display is detected. That indirectly also ensures that Windows 10 is being used and with more or less current video drivers.

**[Always]**: Enable the use of 10-bit swapchain regardless of whether or not there is an HDR display. It is the equivalent setting value that replaces advanced setting `try10bitoutput`. It can also be used with Windows 8, etc. with DirectX feature level >= 11.0. It will not be activated effectively if the system does not support it in any way.

**[Never]**: Never enable 10-bit swapchain for SDR content. Users with an HDR display can use this option for testing purposes or if for some reason they do not want to use 10-bit for SDR content. Even with this setting, 10 bits will still be used when playing content in HDR pass-through mode.

## How has this been tested?
Runtime tested on two systems HDR capable / not capable.

## What is the effect on users?
For users without HDR screen it allows to use 10-bit video surfaces without the need to use advanced settings and for users with HDR screen it allows to disable 10-bit for SDR content.

## Screenshots (if appropriate):

![screenshot00006](https://user-images.githubusercontent.com/58434170/128244225-f15c6a07-795a-462e-b40b-355aaea48f5e.png)

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [X] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

thexai committed:
[Windows] Add new setting "Use 10 bit for SDR" in Display Settings

thexai commented:
@lrusak, please take a look to settings changes.
@jjd-uk, for review xml / strings.po changes.

jjd-uk commented on addons/resource.language.en_gb/resources/strings.po:
```diff
@@ -20786,7 +20810,11 @@ msgctxt "#36577"
+#. Description of setting "System / Display / Use 10-bit surfaces" with label #36095
+#: system/settings/settings.xml
+msgctxt "#36578"
+msgid "Improves video quality using 10-bit video surfaces. YUV to RGB and limited to full range video conversions are more precise with 10 bits reducing quantization errors and banding. Even if video output is limited to 8-bit (by video driver), improvement's in internal video processing are maintained.[CR][Auto] enables 10 bits only if it's certain that the system supports it, although most mode ...
```
If I've understood purpose of the setting correctly then perhaps this could be changed to:  

"Improves video quality by using 10-bit video surfaces for SDR. [CR][Auto] Enables 10 bit for SDR only if the connected display supports HDR. [Always] Enables 10 bit for **SDR** even if the connected display doesn't support HDR. [Never] Do not use 10 bit for SDR.[CR]HDR passthrough mode always uses 10-bit regardless of this setting."

In my view there should just enough to understand what the setting does, no need to go into too much detail.

thexai replied:
Yes, you have understood correctly and I like the idea of making it clear that it refers only to SDR video.

Somehow I am using the term "video surfaces" to make it clear that it is not about the "video output". That is, what this option does is that internally Kodi processes the video in 10 bits but since the output is SDR in most cases it will end up outputting 8 bits video to the TV since it is the default configuration of Windows / video drivers. Even so, the quality improvement obtained is very notable since the internal processing is much more precise with 10-bit.

Maybe text can be:

Improves video quality by using 10-bit video surfaces for Standard Dynamic Range (SDR) video during internal processing.
[Auto] Enables 10-bit for SDR only if the connected display supports HDR.
[Always] Enables 10-bit for SDR even if the connected display doesn't support HDR.
[Never] Do not use 10-bit for SDR.
HDR passthrough mode always uses 10-bit regardless of this setting.

And perhaps add at end:

Note: the video output to TV may be still limited to 8-bit due the video driver settings or hardware supported modes, but it's still an improvement to use 10-bit internally for video range conversions.

jjd-uk commented on addons/resource.language.en_gb/resources/strings.po:
```diff
@@ -18404,7 +18404,31 @@ msgctxt "#36050"
+#. Label of setting "System / Display / Use 10-bit surfaces"
+#: system/settings/settings.xml
+msgctxt "#36095"
+msgid "Use 10-bit surfaces"
```
As this is only for SDR and probably most uses doesn't know what a surface is or used for, perhaps change to something like:

Use for 10 bit for SDR

thexai replied:
Maybe:

Use 10-bit video surfaces for SDR

jjd-uk commented:
A couple of comments on the language strings used. Others however may disagree so perhaps wait for further feedback before you consider changing these strings.

thexai commented:
I've thought better of it and changed the strings to those proposed by @jjd-uk. Simpler is better also thinking about translations.

Also changed the setting value from "Auto" to "Auto detect" as it could be ambiguous (considering translations too).

garbear reviewed (approved):
The diff appears to be exactly the two points described in **Description**. No problems that I can see.

jjd-uk commented:
I see afedchin has now approved. The settings.xml changes look correct but I haven't had chance to check with a test build. If you can wait before merger I'll do it after work today, but I would have no problem if you don't want to wait. As if there was anything that I thought could be handled differently then a follow up is easily done.

thexai commented:
No hurry. I am using it in Windows x64 and everything is correct. I had some doubts about whether the 10-bit setting could also be used on other platforms or if there was already a similar setting that could be used but I don't think so.

jjd-uk commented on addons/resource.language.en_gb/resources/strings.po:
```diff
@@ -18404,7 +18404,31 @@ msgctxt "#36050"
+msgctxt "#36097"
+msgid "Never"
+msgstr ""
+
```
Similar here, Never exist as String 20420 https://github.com/xbmc/xbmc/blob/7a043e7b6e777423f55a029c9d0f6b9c0dca33a3/addons/resource.language.en_gb/resources/strings.po#L13421

```
#. Settings -> PVR -> Playback -> Switch to fullscreen value
#. Value for setting with label #36095 "System / Display / Use 10 bit for SDR"
#: system/settings/settings.xml
msgctxt "#20420"
msgid "Never"
msgstr ""
```

jjd-uk commented on addons/resource.language.en_gb/resources/strings.po:
```diff
@@ -18404,7 +18404,31 @@ msgctxt "#36050"
+msgctxt "#36098"
+msgid "Always"
+msgstr ""
 
```
Similar again. **Always** already exists as String 20422 https://github.com/xbmc/xbmc/blob/7a043e7b6e777423f55a029c9d0f6b9c0dca33a3/addons/resource.language.en_gb/resources/strings.po#L13430

so this becomes

```
#. Value for setting with label #170 "Player / Videos / Adjust display refresh rate"
#. Value for setting with label #36095 "System / Display / Use 10 bit for SDR"
#: system/settings/settings.xml
msgctxt "#20422"
msgid "Always"
msgstr ""
```

jjd-uk commented:
I missed that you had created duplicates of exisitng strings, please reuse existing strings whereever possible if they are used in a similar context. Thanks for all the context for translators, and although these are exisitng strings I've asked you to reuse, it's good practice to update the help context for any existing uses to aid translators of any new languages or simply where they haven't been done yet (unlikely for these ones).

The settings.xml changes look good as is.

thexai commented:
Updated strings with suggested changes and now setting is #36098 so that it follows the last one used.
